vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/occupancy_handler.rb

```diff
@@ -0,0 +1,67 @@
+module SecureRooms
+
+  module AccessHandlers
+
+    class OccupancyHandler
+
+      attr_reader :event
+
+      def self.process(event)
+        new(event).process
+      end
+
+      def initialize(event)
+        @event = event
+      end
+
+      def process
+        event_attribute =
+          if entering?
```

Comment:
Could this be collapsed down into the main conditional below? Having these two sets of conditionals is a little confusing.